### ISSUE TITLE
Add support (or at least namespacing) for requesting more than one metric per request

### ISSUE THREAD

**Issue by MaineC** [elastic/elasticsearch#20365]
This relates to #19195

As most likely the largest part in completing a rank evaluation request is with retrieving results for test queries it might make sense to support computing more than one metric at a time. While we might get away with postponing the complete implementation we should look at request/response formats and add additional name spacing where needed.

**Comment by MaineC** [comment]
So on the request side, we want to expect a list of metrics:

    GET /_rank_eval
    {
        "requests": [
        {
            "id": "JFK query",                              
            "request": {"query": {"match": {"opening_text": {"query": "JFK"}}}},    
            "ratings": [
            {
              "rating": 1,
              "_type": "page",
              "_id": "30900421",
              "_index": "enwiki_rank"
            }],
          "summary_fields": ["title"],                      
        },
        "metrics": {
          [
              "reciprocal_rank": {},
              "precision": {}
        ]}}


That's what the response side looks like:

     {  
       "rank_eval":{  
          "quality_level":[  
             {  
                "name":"precision",
                "value":0.12698412698412698
             },
             {  
                "name":"reciprocal_rank",
                "value":0.5
             }
          ],
          "details":{  
             "0":{  
                "quality_level":[  
                   {  
                      "name":"precision",
                      "value":0.1111111111111111
                   }
                ]
             }
          }
       }
    }

**Comment by MaineC** [comment]
The original question came from discussion with @colings86 and @bleskes - after discussing f2f with @cbuescher summarizing below:

tl;dr: IIRC the original intent was to reserve some namespace for requesting more than one metric at a time to avoid backwards compatibility issues in the future. After looking at the resulting request/response format changes and dependencies introduced between search request and metrics requested this doesn't sound like such a great idea to do in a first iteration anymore, in particular as all it would save is another search round trip.

Looking at the proposed request above the first question that comes to mind is why not give each metric some unique identifier and allow for requesting more than one metric configuration per request as well (e.g. let users ask for Precision_at_10, Precision_at_5 and Precision_at_1). 

In #21310 we decided to remove the separate size configuration for evaluation metrics and rely only on the size configuration the user set in the rated search request to avoid confusion between the two. So thinking the idea of requesting metrics with different configuration further means re-introducing that parameter per metric, and relying on the user to make sure to request enough search results to cover for the largest above configuration. This sounds like an awesome way to screw things up.

Unless I'm missing something the advantage of allowing for more than one metric would be to execute the rank evaluation request just once. By extension this means executing all rated search requests contained just once.

Given that this is an API that is in no way real time, I'm wondering whether these savings are large enough to justify adding more potential for confusion to the request and increasing the size of the request/response themselves.

**Comment by cbuescher** [comment]
> Given that this is an API that is in no way real time, I'm wondering whether these savings are large enough to justify adding more potential for confusion to the request and increasing the size of the request/response themselves

I'm opting for calculating only one metric per request, if more metrics or different parametrizations are needed, the client can issue multiple requests.
